### PR TITLE
Add Pangea4 host configs

### PR DESCRIPTION
**Pull request by Bubusch**
Adds the Pangea 4 host config files :

| host-config   | Description | Compiler | MPI | BLAS-LAPACK |
| -------- | ------- | ------- | ------- | ------- |
| pangea4-base.cmake  | base configuration for Pangea4 (CPU cluster), all libs enabled, no openMP  | X | X | X |
| pangea4-gcc12.1-hpcxompi2.17.1-onemkl2023.2.0 | default gcc build | cray wrappers for gnu (cc, CC, ftn)  | HPC-X OpenMPI | oneAPI MKL |
| pangea4-gcc12.1-openmpi4.1.6-onemkl2023.2.0 | alternative gcc build | cray wrappers for gnu (cc, CC, ftn)  | OpenMPI | oneAPI MKL |
| pangea4-gcc12.1-openmpi4.1.6-openblas0.3.23 | alternative gcc build | cray wrappers for gnu (cc, CC, ftn)  | OpenMPI | OpenBLAS |
| pangea4-gcc12.1-hpcxompi2.17.1-openblas0.3.23 | alternative gcc build | cray wrappers for gnu (cc, CC, ftn)  | HPC-X OpenMPI | OpenBLAS |

Linked to GEOS' PR [#3081](https://github.com/GEOS-DEV/GEOS/pull/3081)

### CODE REVIEW

**Review by jhuang2601 — APPROVED** [review]
But I think this one is not needed any more, as long as PR (#3081) is merged!

**Review by MelReyCG — APPROVED** [review]

File: host-configs/TOTAL/pangea4-base.cmake [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#
+#######################################
+
+#######################################
+# SCIENTIFIC LIBRARIES
+#######################################
+
+# build all TPLs scientific libs and let GEOS build choose which one to use
+
+set( ENABLE_FESAPI      ON CACHE BOOL "" FORCE )
```
Comment (untereiner) [inline host-configs/TOTAL/pangea4-base.cmake:18]: You can remove ENABLE_FESAPI

**Comment by CusiniM** [comment]
is there any way to create a CI job to mimic this system?

**Comment by jhuang2601** [comment]
> is there any way to create a CI job to mimic this system?

That will be great.

**Comment by TotoGaz** [comment]
I think @Bubusch is working on it?

**Comment by Bubusch** [comment]
@CusiniM Yes CI job is in our todo list. I created a small epic to regroup our tasks related to the builds on TTE infra -> https://github.com/GEOS-DEV/GEOS/issues/3172

**Pull request merged**